### ISSUE TITLE
Sound sonar makes for players doesn't change for directional sonar

### ISSUE THREAD

**Issue by GoodDays13** [Regalis11/Barotrauma#10291]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

When the sonar is in directional ping mode, to other sonars, and to the sonar itself, it still makes the same sound as full sonar. This would affect pvp, and affect people like me who try to find out game mechanics by testing in-game.

### Reproduction steps

Start a game so you have a sub and some way to swim away from it.
Turn on directional sonar and remember the direction.
![image](https://user-images.githubusercontent.com/42877449/200629128-23d996db-f85e-48c7-9e2d-0365905246bb.png)
Get a diving suit and handheld sonar and start swimming away from the sub in the other direction than you set the directional ping.
It is still visible the same distance as it would with 360 ping.
![image](https://user-images.githubusercontent.com/42877449/200628811-54948394-2ef7-4816-ad3c-b021b5a14c7c.png)

You can also use it with passive sonar and see it bounce off a lot of things around you.
![image](https://user-images.githubusercontent.com/42877449/200630715-ce4d4882-df4e-45df-9f18-8486145a5b29.png)

### Bug prevalence

Happens every now and then

### Version

0.19.14.0

### Which operating system did you encounter this bug on?

Windows

**Comment by NilanthAnimosus** [comment]
Tested against release commit https://github.com/Regalis11/Barotrauma-development/commit/2b57faa07493e732f98261ead832ba904168f57b

Not quite sure how to explain this other than it doesn't look right, The sonar was picking up sounds seemingly erratically (I imagine mostly from the reactor here), but directional sonar didn't seem to either be present, or was not working as intended. also none of the terrain was visible on the portable sonar outside the submarine using directional sonar.

area around submarine (for video):
![image](https://user-images.githubusercontent.com/29177976/204504853-701ad98e-ca35-4b6f-a428-f701a74b8b98.png)

Heres a video since my words are likely lacking here:

https://user-images.githubusercontent.com/29177976/204504330-e14c4cb5-4dae-4db1-872b-288bfbd3aa21.mp4

**Comment by Regalis11** [comment]
Pushed some further fixes in https://github.com/Regalis11/Barotrauma-development/commit/499bc8badcdb732d6a35e68a037ca99417d8c2f1

Fixed some issues in https://github.com/Regalis11/Barotrauma-development/commit/30724589cb48a4045d7c8721f6dfaa885b6cbe4f (prevBlips list never being filled, leftover debug code).
Fixed blip on directional sound sources appearing very irregularly, because we checked if it was inside it's own sector.
Adjusted ranges of the fake "pings" sound sources make to reveal thigns around them. Previously they were clamped to 50% of the sonar range, no matter how loud they were, meaning that you e.g. sometimes wouldn't be able to see a submarine's pings when using a handheld sonar outside, even if the range of the sub's ping should be more than enough to reach you.
